shopify-client/src/components/ShopForm.vue

```diff
@@ -63,9 +170,48 @@
 </template>
 
 <script>
+
+let addDiff = (newAttributes, target) =>
+    newAttributes.forEach(function (attribute) {
+      if(target.every(element => attribute.name !== element.name)){
+        target.push(attribute)
+      }
+    });
+
 export default {
   name: "ShopForm",
   props: ["handleSubmit", "initialValues"],
+  created: async function() {
+    await fetch('/api/category/list')
+        .then((response) => response.json())
+        .then((response) => {
+          if (response.status == 404) {
+            throw new Error(response.message);
+          }
+          return response;
+        })
+        .then((data) => {
+          addDiff(data.filter(category => {
```

Comment:
What happens if we simply replaced the `this.categories` or `this.tags` with `this.initialValues.categories`/`this.initialValues.tags` instead of manually adding the difference? It looks like `this.initialValues.categories/tags` already has all categories/tags we need.

Reply:
I think you misunderstand what this block does. This adds to this.categories the categories that aren't in currently in the shop. Ditto for tags.

This facilitates the addition of existing tags.

Reply:
3fd4fa0 - added some comments that should explain the block.